### PR TITLE
Simplify Snapshot structure

### PR DESCRIPTION
**Pull request by ip1981**
Don't need detailed views

Signed-off-by: Igor Pashev <pashev.igor@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2222)

### CODE REVIEW

File: iml-wire-types/src/snapshot.rs [hunk 1]
```diff
@@ -30,28 +26,11 @@ pub struct List {
...
-    /// E. g. MDT0000
-    pub snapshot_role: Option<String>,
-    /// Filesystem id (random string)
+    /// Snapshot filesystem id (random string)
     pub snapshot_fsname: String,
     pub modify_time: DateTime<Utc>,
     pub create_time: DateTime<Utc>,
-    /// Snapshot status (None means unknown)
-    pub status: Option<Status>,
+    pub mounted: Option<bool>,
```
Comment (jgrund) [inline iml-wire-types/src/snapshot.rs:33]: In what case do we get an option here?

File: iml-agent/src/action_plugins/lustre/snapshot/parse.rs [hunk 2]
```diff
@@ -69,47 +63,35 @@ fn parse_date(mut s: String) -> Result<DateTime<Utc>, &'static str> {
...
 
     let modify_time = parse_date(modify_time_s)?;
     let create_time = parse_date(create_time_s)?;
-    let status = match status_s.as_str() {
-        "mounted" => Some(Status::Mounted),
-        "not mount" => Some(Status::NotMounted),
+    let mounted = match status.as_str() {
+        "mounted" => Some(true),
+        "not mount" => Some(false),
         _ => None,
```
Comment (jgrund) [inline iml-agent/src/action_plugins/lustre/snapshot/parse.rs:80]: Is there a third state? I think there is only two

Comment (jgrund) [inline iml-agent/src/action_plugins/lustre/snapshot/parse.rs:80]: Three states:

https://github.com/whamcloud/lustre/blob/3e9f1b164be9be4543bedd29fcdf728f63303540/lustre/utils/lsnapshot.c#L2005-L2011

Comment (ip1981) [reply iml-agent/src/action_plugins/lustre/snapshot/parse.rs]:
```rust
    let mounted = match status.as_str() {
        "mounted" => Some(true),
        "not mount" => Some(false),
        "unknown" => None,
        _ => return Err("mounted, not mount, unknown"),
    };

```

**Review by mkpankov — APPROVED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2222)* status: all files reviewed, 1 unresolved discussion (waiting on @jgrund)

**Comment by jgrund** [comment]
This has conflicts on master.

**Review by mkpankov — APPROVED** [review]
> Reviewed 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2222)* status: all files reviewed, 1 unresolved discussion (waiting on @jgrund)

**Review by jgrund — APPROVED** [review]

**Pull request merged**